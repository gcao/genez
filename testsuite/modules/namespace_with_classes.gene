#!/usr/bin/env gene run

# Test namespace declarations with classes

# Namespace with class definition
(ns geometry
  (class Point
    [x 0]
    [y 0]
    
    (.fn distance-to [other]
      (do
        (var dx (- other/x /x))
        (var dy (- other/y /y))
        (+ (* dx dx) (* dy dy)))))
  
  (class Line
    [start]
    [end]
    
    (.fn length []
      (start .distance-to end))))

# Namespace with multiple classes
(ns ui/components
  (class Button
    [text ""]
    [enabled true])
    
  (class Label
    [text ""]
    [visible true]))

(print "Namespace with classes parsed successfully")