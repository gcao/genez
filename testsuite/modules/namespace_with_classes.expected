Namespace with classes parsed successfully